pkg/registry/controller/etcd/etcd.go

```diff
@@ -46,12 +48,14 @@ func NewREST(h tools.EtcdHelper) *REST {
 		// Produces a path that etcd understands, to the root of the resource
 		// by combining the namespace in the context with the given prefix
 		KeyRootFunc: func(ctx api.Context) string {
-			return etcdgeneric.NamespaceKeyRootFunc(ctx, controllerPrefix)
+			prefix := path.Join("/", h.PathPrefix, controllerPrefix)
```

Comment:
This is the wrong place for this change, this is too high level. The only change in this file should be the change you made on line 39 above.